src/status_im/extensions/registry.cljs

```diff
@@ -143,22 +143,18 @@
                                      :ephemeral? ephemeral?}
                         :active? true}
         new-extensions (assoc (:extensions account) url extension)]
-    (if ephemeral?
-      (fx/merge cofx
-                {:dispatch (if modal?
-                             [:navigate-back]
-                             [:navigate-to-clean :my-profile])}
-                (when on-activation (on-activation)))
-      (fx/merge cofx
-                {:utils/show-popup {:title      (i18n/label :t/success)
-                                    :content    (i18n/label :t/extension-installed)
-                                    :on-dismiss #(re-frame/dispatch (if modal?
-                                                                      [:navigate-back]
-                                                                      [:navigate-to-clean :my-profile]))}}
-                (when hooks (accounts.update/account-update {:extensions new-extensions} {}))
-                (when hooks (add-to-registry url extension-data true))
-                (when on-installation (on-installation))
-                (when on-activation (on-activation))))))
+    (do
+      (re-frame/dispatch (if modal?
```

Comment:
Can't we still have regular fx usage here?

Reply:
I tried first but the sequence was not correct: i get navigated to extension screen and then the other navigation event arrives moving back to profile. In this way the it seems to work (go back to profile and thennavigate to extension screen arrives).. i dont know if there is a better way to achive the same result

Reply:
Hmm then it means we rely on timing (even with your option) which is not good. Depending on phone speed you can get weird results. 
Do you think we could have some kind of callback mechanism? e.g. the first event fires itself the second event, thus controlling the flow?

Reply:
I don't know really how it works behind the scene here, but I was thinking there is a queue of events and in this way i'm sure I put the events in the correct order.. but if it is not the case, totally agree with you in case it actually depends on timings..

Reply:
You can find some details here: https://github.com/Day8/re-frame/blob/master/docs/Effects.md#order-of-effects

Reply:
ok, right :-)
this evening i'll come with a a better solution!

Reply:
need to study carefully this too https://github.com/Day8/re-frame/blob/master/docs/EventHandlingInfographic.md

Reply:
don't know too much context.. although I'm working on a bounty that requires full screen overlay so was looking into this for inspiration.. fwiw thought I'd chime in and share that:

in re-frame there is also `:dispatch-later` and `:dispatch-n`

Reply:
yep.. have to do some tests with `:dispatch-n` which should preserve event ordering...but i still have to study\understand if event order guarantee is enough here to have a correct behaviour

Reply:
really need to put event\fx processing in debug mode to see what happens :)

Reply:
Going off for vacation... I'll continue next week!

Reply:
event ordering is not guaranteed with dispatch-n. If you want ordering either use merge-fx (changes to app-db will apply in order, but not side effects), or dispatch the next event in the current one if you want sequential side effects

Reply:
@yenda so you're saying that's ok for you the current version where i'm dispatching the event before returning effects? moreover it seems to me that `(on-*)` lifecycle are returning `nil` so i think they are directly dispatching events instead of generating effects.

So `:navigate-back` event is guaranteed to run before the event dispatched by `(on-installation)` \ `(on-activation)`, right?

Reply:
i was playing with a solution like:

```
(fx/defn identity-fx
  [_ cb]
  (when cb {::identity-event {:cb cb}}))

(fx/defn install
  [{:keys [db] :as cofx} url {:keys [hooks] :as extension-data} modal?]
  (let [{:account/keys [account]} db
        on-installation (get-in extension-data [:lifecycle :on-installation])
        on-activation   (get-in extension-data [:lifecycle :on-activation])
        ephemeral?      (get-in extension-data [:lifecycle :ephemeral])
        extension       {:id        url
                         :name      (or (get-in extension-data [:meta :name]) "Unnamed")
                         :url       url
                         :lifecycle {:on-installation   on-installation
                                     :on-deinstallation (get-in extension-data [:lifecycle :on-deinstallation])
                                     :on-activation     on-activation
                                     :on-deactivation   (get-in extension-data [:lifecycle :on-deactivation])
                                     :ephemeral?        ephemeral?}
                         :active?   true}
        new-extensions  (assoc (:extensions account) url extension)]
    (if ephemeral?
      (fx/merge cofx
                (when on-activation (on-activation)))
      (fx/merge cofx
                 {:dispatch (if modal?
                              [:navigate-back]
                              [:navigate-to-clean :my-profile])}
                (identity-fx on-installation)
                (identity-fx on-activation)
                (when hooks (accounts.update/account-update {:extensions new-extensions} {}))
                (when hooks (add-to-registry url extension-data true))))))
```
to be adjusted but it gives the idea.. don't know if it is better than the current solution.. i'm actually postponing the on-* execution to fx processing

Reply:
Ping @flexsurfer for ideas

Reply:
sorry for the delay, @tbenr  have you looked to https://github.com/status-im/status-react/blob/6b99225e0680c176ad424a168fa534864d4d68df/src/status_im/ui/screens/navigation.cljs#L117 and https://github.com/status-im/status-react/blob/6b99225e0680c176ad424a168fa534864d4d68df/src/status_im/ui/screens/navigation.cljs#L17